fetch_subprojects.py

```diff
@@ -0,0 +1,75 @@
+#! /usr/bin/env python
+"""
+A script to generate a flat subproject directory from a tree of dependency directories
+"""
+import os
+import subprocess
+import textwrap
+
+def dependency_directory():
+    """
+    A function isolating the path specification to the project's dependency directory
+    """
+    return os.path.join( os.getcwd(), "dependencies" )
+
+def clone_submodule( relative_path ):
+    """
+    A function to clone a git submodule
+    """
+    print( textwrap.dedent(
+      """
+      ----------------------------------------
+      Fetching {relative_path}...
+      ----------------------------------------
+      """.format( relative_path = relative_path ) ) )
+    invocation = [ "git", "submodule", "update", "-q","--init", "--", relative_path ]
+    if os.name == "nt":
+        invocation.insert( 0, "powershell" )
+
+    clone = subprocess.Popen( invocation )
+    clone.communicate()
+
+def update_repository():
+    """
+    A function to update a submodule to lastest commit of the master branch
+    """
+    print("Updating to master branch...")
```

Comment:
You could do:  
    
    print("Updating to master branch...\n")

instead of two print statements.